app/model/package.scala

```diff
@@ -45,14 +72,34 @@ case class UploadedFile(
   downloadUrl: URL,
   size: Long,
   uploadDetails: UploadDetails,
-  requestContext: RequestContext
-)
+  requestContext: RequestContext,
+  override val userMetadata: Map[String, String]) extends UserMetadataLike {
+
+  def copyWithUserMetadata(kv: (String,String)): UploadedFile = {
+    copy(userMetadata = (this.userMetadata + kv))
+  }
+
+  def copyWithUserMetadata(kv1: (String,String), kv2: (String,String), kv3: (String,String)*): UploadedFile = {
+    copy(userMetadata = (this.userMetadata + (kv1, kv2, kv3:_*)))
+  }
+}
 
 case class QuarantinedFile(
   callbackUrl: URL,
   reference: FileReference,
   error: ErrorDetails,
-  requestContext: RequestContext)
+  uploadDetails: UploadDetails,
+  requestContext: RequestContext,
+  override val userMetadata: Map[String, String]) extends UserMetadataLike {
+
+  def copyWithUserMetadata(kv: (String,String)): QuarantinedFile = {
+    copy(userMetadata = (this.userMetadata + kv))
+  }
+
+  def copyWithUserMetadata(kv1: (String,String), kv2: (String,String), kv3: (String,String)*): QuarantinedFile = {
+    copy(userMetadata = (this.userMetadata + (kv1, kv2, kv3:_*)))
+  }
+}
```

Comment:
That would certainly be nicer. But I don't think it would play nicely with the underlying Map call:
https://www.scala-lang.org/api/2.11.12/index.html#scala.collection.immutable.MapLike@+[B1%3E:B](elem1:(A,B1),elem2:(A,B1),elems:(A,B1)*):scala.collection.immutable.Map[A,B1]